### PR TITLE
Fix: SQL syntax in editor

### PR DESCRIPTION
**Pull request by mykalmax**
refactored code editor syntax highlighting

### CODE REVIEW

File: web/client/src/workers/sqlglot/sqlglot.py [hunk 1]
```diff
@@ -34,4 +34,30 @@ def get_dialect(name: str = "") -> str:
...
+def validate(sql: str = "", read: DialectType = None) -> str:
+    is_valid = True
+    try:
+        sqlglot.transpile(
+            sql, read=read, pretty=False, unsupported_level=sqlglot.errors.ErrorLevel.IMMEDIATE
+        )
+    except sqlglot.errors.ParseError as e:
+        is_valid = False
+
+    return json.dumps(is_valid)
```
Comment (vchan) [inline web/client/src/workers/sqlglot/sqlglot.py:54]:
```suggestion
    try:
        sqlglot.transpile(
            sql, read=read, pretty=False, unsupported_level=sqlglot.errors.ErrorLevel.IMMEDIATE
        )
    except sqlglot.errors.ParseError as e:
        return json.dumps(False)  # or just return 'false'
    return json.dumps(True)
```

File: web/client/src/library/components/editor/extensions/SQLMeshDialect.ts [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  options?: { types: string; keywords: string },
+  dialects?: string[],
+) => LanguageSupport
+
+export const SQLMeshDialect: ExtensionSQLMeshDialect = function SQLMeshDialect(
+  models,
+  options = { types: '', keywords: '' },
+  dialects,
+): LanguageSupport {
+  const SQLKeywords = options.keywords + WHITE_SPACE
```
Comment (vchan) [inline web/client/src/library/components/editor/extensions/SQLMeshDialect.ts:26]: Can you please remind me again why all these strings need a trailing white space?

File: web/client/src/library/components/editor/extensions/SQLMeshDialect.ts [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+  sqlglotWorker.removeEventListener('message', handler)
+}
+
+export function getSQLMeshModelKeywords(
+  dialects: string[] = [],
+): Map<string, Completion[]> {
+  return new Map([
+    [
+      'kind',
```
Comment (vchan) [inline web/client/src/library/components/editor/extensions/SQLMeshDialect.ts:128]: Should we include EMBEDDED here?

File: web/client/src/utils/index.ts [hunk 4]
```diff
@@ -48,11 +48,11 @@ export function isObjectLike(value: unknown): boolean {
   return isNotNil(value) && typeof value === 'object'
 }
 
-export function isNil(value: unknown): boolean {
+export function isNil(value: unknown): value is undefined | null {
   return value == null
 }
 
-export function isNotNil(value: unknown): boolean {
+export function isNotNil<T>(value: T | null | undefined): value is T {
```
[the remark this answers is not in the document]
Comment (mykalmax) [reply web/client/src/utils/index.ts:55]: It means that value is not null / undefined but actually some value of type T

[the remark this answers is not in the document]
Comment (vchan) [reply web/client/src/utils/index.ts:55]: Okay, but isn't that where the return type goes?

[the remark this answers is not in the document]
Comment (mykalmax) [reply web/client/src/utils/index.ts:55]: Oh, yes it is return type.
this should let eslint/typescript know that value we checking exists for code block below the check

File: web/client/src/library/components/editor/extensions/SQLMeshDialect.ts (hunk 2 above)
Comment (mykalmax) [reply web/client/src/library/components/editor/extensions/SQLMeshDialect.ts:26]: in case `options.keywords` coming without trailing space

Comment (vchan) [reply web/client/src/library/components/editor/extensions/SQLMeshDialect.ts:26]: Is it for concatenating with `SQLMeshKeywords` below for `lang`? Does codemirror expect a trailing whitespace? What would happen if `SQLKeywords` and `SQLMeshKeywords` didn't have trailing whitespaces, but instead below is `keywords: (SQLKeywords + WHITE_SPACE + SQLMeshKeywords).toLowerCase()`?

**Review by vchan — APPROVED** [review]

**Pull request merged**